### ISSUE TITLE
Allowed unsecured hosts list.

### ISSUE THREAD

**Issue by HighSoftWare96** [apache/cordova-android#812]
# Feature Request

## Motivation Behind Feature

We should be able to add a list of unsecure network hosts in order to allow unsecured network requests for there hosts. Right now in android Ionic allows to send requests to unsecured or self-certified hosts only in debug mode and not in release mode.

## Feature Description

In this file we should access to this list (configured somewhere, maybe in the manifest.xml file) before calling the `super.onReceivedSslError(view, handler, error);` so the developers can define which hosts can be used without SSL trust.
https://github.com/apache/cordova-android/blob/bd1697dbd27411d8b12d9ad0902babaaae327efa/framework/src/org/apache/cordova/engine/SystemWebViewClient.java#L225

## Alternatives or Workarounds

Right now the only workaround is to change the actual Java code to call the next handler instead of the `super.onReceivedSslError(view, handler, error);` in any cases, as it is explained here: http://ivancevich.me/articles/ignoring-invalid-ssl-certificates-on-cordova-android-ios/.

**Comment by breautek** [comment]
May I ask why you would want such a feature for release mode?

**Comment by HighSoftWare96** [comment]
For example when I have to create a production build that uses development servers or API I should be able to add my domains in the trusted ones. This feature is even included in iOS 9 within the Info.plist.

**Comment by breautek** [comment]
For iOS, it is possible because cordova offers an ability to modify the plist file via `<edit-config>`. This doesn't mean it's a good idea (in my opinion...)

Allowing this kind of configuration on release mode could trigger Google Play's security detection algorithm as APKs uploaded to the google play store is automatically scanned for potential security issues as described [here](https://developer.android.com/google/play/asi). I am assuming you want a release mode so that you can use Google Play as a distributor for your testers.

Because of this, I don't think I can say I'm in favour of this request. The alternatives would be to install a verified cert on your staging server or distribute a debug APK that your testers can side-load.

I should also mention that I don't represent Cordova, and this is just my personal opinion.

Should this feature request be fulfilled, then this [documentation](https://developer.android.com/training/articles/security-ssl#UnknownCa) will likely help implement it in a secure manner, which involves providing a set of CAs that are considered trustworthy, rather than simply accepting domains which can easily be vulnerable to DNS trickery.

**Comment by HighSoftWare96** [comment]
Sure, if you are interested in this functionality.. I've seen a lot of articles in which this feature may be willed

**Comment by HighSoftWare96** [comment]
Any news on this? I've seen for example this https://cordova.apache.org/docs/en/latest/reference/cordova-plugin-whitelist/ plugin but for me that I'm using Ionic Cordova I'm not able to make the network requests I've intended. I've also tried by inserting the self-signed certificate in the Android project but without any effects. The only solution remains actually this http://ivancevich.me/articles/ignoring-invalid-ssl-certificates-on-cordova-android-ios/.
